hack/tools/tilt-prepare/main.go

```diff
@@ -184,21 +227,43 @@ func tiltResources(ctx context.Context) error {
 	// Add a provider task for each name/path defined using the --provider flag.
 	for _, p := range *providersBuildsFlag {
 		pValues := strings.Split(p, ":")
-		if len(pValues) != 2 && len(pValues) != 3 {
-			return errors.Errorf("[resources] failed to parse --provider flag %s: value should be in the form of name:path[:debug]", p)
+		if len(pValues) != 2 {
+			return errors.Errorf("[resources] failed to parse --provider flag %s: value should be in the form of name:path", p)
 		}
 		name := pValues[0]
 		path := pValues[1]
-		debug := false
-		if len(pValues) == 3 && pValues[2] == "debug" {
-			debug = true
-		}
-		tasks[name] = providerTask(fmt.Sprintf("%s/config/default", path), fmt.Sprintf("%s.provider.yaml", name), debug)
+		tasks[name] = providerTask(name, fmt.Sprintf("%s/config/default", path), setDebugDefaults(ts))
```

Comment:
What about moving the call to `setDebugDefaults` into `readTiltSettings`, so the latter returns a ready-to-use object?

Reply:
+1

Reply:
Absolutely. Changed it accordingly.